pybigtools/src/lib.rs

```diff
@@ -879,15 +879,34 @@ impl BigBedWrite {
 }
 
 enum BigWigAverageOverBedEntriesIteratorRet {
-    Single(f64),
-    WithName((String, f64)),
+    All((u32, u32, f64, f64, f64, f64, f64)),
+    Float64(f64),
+    UInt32(u32),
+    NameAndAll((String, u32, u32, f64, f64, f64, f64, f64)),
+    NameAndFloat64((String, f64)),
+    NameAndUInt32((String, u32)),
+}
+
+enum BigWigAverageOverBedStatistics {
```

Comment:
My first thought was that this seems unnecessary, and that it makes more sense to just always return all stats in a dict. But, if you only want one stat, it is probably better to not allocate the dict.

Reply:
I wonder if returning a Namedtuple would be better than a dict.

Reply:
A namedtuple would be good too!

Reply:
Do you know how to create it with pyo3?

Reply:
It's a good question.

Looking at https://github.com/PyO3/pyo3/issues/1647, it looks like this probably isn't implemented. There is `PyStructSequence`, but only the ffi layer seems available for use. 

Testing a bit; it may be possible to essentially fake this by calling into python directly via `.call`s

Reply:
Looks like this should work to make a namedtuple (roughly):

To add to the module:
```rust
let collections = PyModule::import(py, "collections").unwrap();
let namedtuple = collections.getattr("namedtuple").unwrap();
let summary_statistics = namedtuple.call(("SummaryStatistics".to_object(py), ("start", "end", "mean").to_object(py)), None).unwrap();m.add("SummaryStatistics", summary_statistics).unwrap();
```

And then to use:
```rust
let summary_statistics = module.getattr("SummaryStatistics").unwrap();
let val = summary_statistics.call((0, 0, 0.0), None).unwrap();
```